saga/inMemorySagaLog.go

```diff
@@ -0,0 +1,71 @@
+package saga
+
+import "errors"
+import "fmt"
+import "sync"
+
+/*
+ * In Memory Implementation of a Saga Log, DOES NOT durably persist messages.
+ * This is for local testing purposes.
+ */
+type inMemorySagaLog struct {
+	sagas map[string][]sagaMessage
+	mutex *sync.RWMutex
+}
+
+/*
+ * Returns an Instance of a Saga based on an InMemorySagaLog
+ */
+func InMemorySagaFactory() Saga {
```

Comment:
I'll change to Make for now and then if we want to align on New we can do a refactor!